lib/ecto_ldap/adapter.ex

```diff
@@ -295,10 +322,23 @@ defmodule Ecto.Ldap.Adapter do
     split_and_not_nil(t, count - 1, false, [h|acc])
   end
 
+  def update(repo, schema_meta, fields, filters, _autogen_id, returning, options) do
+    case update(schema_meta, fields, filters) do
+      :ok ->
+        {:ok, fields}
+      error ->
+        IO.inspect error
+        {:invalid, []}
+    end
+  end
+
   def load(:id, value), do: {:ok, value}
   def load(:string, value), do: {:ok, convert_from_erlang(value)}
+  def load({:array, :string}, []), do: {:ok, []}
+  def load({:array, :string}, value), do: {:ok, convert_from_erlang(value)}
 
-  def dump(:string, value), do: {:ok, convert_to_erlang(value)}
+  def dump(:string, value) when is_list(value), do: {:ok, convert_to_erlang(value)}
+  def dump(:string, value), do: {:ok, [convert_to_erlang(value)]}
```

Comment:
Ok - this code is terrible.

Here's why:

`:eldap.mod_replace/3` takes a list as its last argument.
`:eldap.modify/3' *also* takes a list as its last argument.

In order to support attributes which have a `list` value in LDAP, we needed to be able to distinguish between a character list and an actual list type.  I couldn't find a graceful way to do this before Ecto invoked the `dump` command, so I opted to force all `dump` calls to return erlang-formatted data in list format.

Once the data is received and processed by the `:eldap` calls, no further erlang tranlsation steps would be necessary.  This had an unfortunate side-effect of forcing *all* data being provided to `update` to being in a list format, hence why I'm "un-boxing" the `base` on line 54.

I am certain that there is a better way to handle this, so please let me know if you can think of any boundary cases we should consider, or perhaps a more thoughtful approach with respect to how we interpret incoming data to these methods.